modules/Link.js

```diff
@@ -62,6 +77,63 @@ const Link = React.createClass({
     }
   },
 
+  getInitialState() {
+    return {
+      isActive: isLinkActive(this.context.router, this.props)
+    }
+  },
+
+  componentWillMount() {
+    const { router } = this.context
+    if (router) {
+      this.updateIsActive()
+    }
+  },
+
+  componentDidMount() {
+    const { router } = this.context
+    if (router) {
+      this._unlisten = router.listen(() => {
+        if (this._unlisten) {
+          this.updateIsActive()
+        }
+      })
+    }
+  },
+
+  componentWillReceiveProps(nextProps) {
+    const { router } = this.context
+    if (router) {
+      if (
+        nextProps.to !== this.props.to ||
+        nextProps.onlyActiveOnIndex !== this.props.onlyActiveOnIndex ||
+        nextProps.activeClassName !== this.props.activeClassName ||
+        nextProps.activeStyle !== this.props.activeStyle
+      ) {
+        this.updateIsActive(nextProps)
+      }
+    }
+  },
+
+  componentWillUnmount() {
+    if (this._unlisten) {
+      this._unlisten()
+      this._unlisten = null
+    }
+  },
+
+  updateIsActive(props = this.props) {
+    const { router } = this.context
+    const isActive = isLinkActive(router, props)
+
+    // The code is written this way to avoid wasted
+    // setState() calls that get expensive in large trees.
```

Comment:
(Previously, it would be called during `render`, but now it doesn’t, so it seems like the number of *useful* times is equivalent. As far as I can see any other wins from not calling are the bug, not the feature 😄 )